### PR TITLE
BaseBuilder Corrections

### PR DESCRIPTION
**Pull request by atishamte**
Some BaseBuilder corrections like,
- Undefined variables defined.
- Removed unnecessary declarations
- Typos corrections

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPdocs
- [x] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Database/BaseBuilder.php [hunk 1]
```diff
@@ -390,13 +397,18 @@ public function selectSum($select = '', $alias = '')
...
 	{
-		if (! is_string($select) || $select === '')
+		if ($select === '')
 		{
 			throw new DatabaseException('The query you submitted is not valid.');
 		}
 
+		if (strpos($select, ',') !== false)
+		{
+			throw new DatabaseException('Only single column allowed.');
```
Comment (lonnieezell) [inline system/Database/BaseBuilder.php]: Please add this as a new static method in the DatabaseException like the others. Also the message should be localized and stored in a lang file.

File: system/Database/BaseBuilder.php [hunk 2]
```diff
@@ -2343,13 +2360,11 @@ protected function _truncate($table)
...
 	 */
-	public function getCompiledDelete($reset = true)
+	public function getCompiledDelete(bool $reset = true): string
 	{
 		$table = $this->QBFrom[0];
 
-		$this->returnDeleteSQL = true;
-		$sql                   = $this->delete($table, '', null, $reset);
-		$this->returnDeleteSQL = false;
+		$sql = $this->delete($table, '', null, $reset);
```
Comment (lonnieezell) [inline system/Database/BaseBuilder.php]: While I agree with removing the wrapping lines as not neccessary - the default value for `$returnSQL` in the `delete()` method is false, and this function is expecting it to be returned. 

You need to pass `true` as the 4th argument here.

File: system/Database/BaseBuilder.php [hunk 3]
```diff
@@ -2625,18 +2638,8 @@ protected function compileWhereHaving($qb_key)
 						continue;
 					}
 
-					// $matches = array(
```
Comment (lonnieezell) [inline system/Database/BaseBuilder.php]: Put these back in please. They serve as a nice reference when working on that code...

File: system/Database/Postgre/Builder.php [hunk 4]
```diff
@@ -351,15 +351,15 @@ protected function _truncate($table)
...
-	 * @param boolean     $insensitiveSearch
+	 * @param string    $prefix
+	 * @param string    $column
+	 * @param string    $not
+	 * @param string    $bind
+	 * @param boolean   $insensitiveSearch
 	 *
 	 * @return string     $like_statement
 	 */
-	public function _like_statement(string $prefix = null, string $column, string $not = null, string $bind, bool $insensitiveSearch = false): string
```
Comment (lonnieezell) [inline system/Database/Postgre/Builder.php]: What's the reason for removing null values here? These have been this way for many, many years as part of CI3 and probably before. Unless you've confirmed that every use case for this method can work when those are not null, they need to stay. 

I get they don't follow best practices of having nullable arguments last, but if that's the only reason you're changing it, it needs to be fixed.

Comment (atishamte) [reply system/Database/Postgre/Builder.php]: This method is called only from only one place, from where all the params are getting passed to it.
So there is no need of declaring arguments as optional. Hence removed

**Comment by atishamte** [comment]
@lonnieezell, Changes are done.

**Comment by atishamte** [comment]
Any update on this @lonnieezell

**Comment by atishamte** [comment]
@lonnieezell @jim-parry, why the last commit didn't build by Travis CI?

**Comment by jim-parry** [comment]
There are conflictrs that need to be resolved, would be my guess.

**Comment by atishamte** [comment]
Thanks for starting the build. I have done with all the changes.

**Pull request merged**